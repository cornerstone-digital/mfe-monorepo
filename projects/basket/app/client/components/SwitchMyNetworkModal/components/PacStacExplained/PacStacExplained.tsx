import List from '@web-core/components/atoms/List'
import Section from '@web-core/components/atoms/Section'
import Paragraph from '@web-core/components/atoms/Paragraph'
import BlockContainer from '@web-core/components/molecules/BlockContainer'
import Icon from '@web-core/components/atoms/Icon'
import Heading from '@web-core/components/atoms/Heading'

import styles from './PacStacExplained.scss'

const pacListItems = [
  { id: '1', content: 'Cancel your contract with your current network' },
  { id: '2', content: 'Move your current mobile number over to Vodafone' },
  { id: '3', content: 'Example code: ABC123456' },
]

const stacListItems = [
  { id: '4', content: 'Cancel your contract with your current network' },
  { id: '5', content: "Disconnect your current mobile number. You'll get a new number with Vodafone" },
  { id: '6', content: 'Example code: 123456ABC' },
]

const PacStacExplained = (): JSX.Element => {
  return (
    <BlockContainer
      heading={{ text: 'What are PAC and STAC codes?', size: 6, fontWeight: 'bold' }}
      collapse={{ collapsable: true, collapsed: true, icon: { justify: 'left', appearance: 'dark' } }}
      borderAppearance="none"
      marginBottom={2}
    >
      <div className={styles['content']}>
        <Icon name="info-circle" appearance="dark" size={2} />
        <Section paddingLeft={2}>
          <Heading level={3} size={6} fontWeight="bold">
            Use a PAC code to:
          </Heading>
          <List items={pacListItems} />
          <Heading level={3} size={6} fontWeight="bold">
            Use a STAC code to:
          </Heading>
          <List items={stacListItems} />
          <Heading level={3} size={6} fontWeight="bold">
            How do I get my PAC or STAC?
          </Heading>
          <Paragraph>Just send a free text to 65075 from the mobile you want to switch</Paragraph>
          <Paragraph>
            For <strong>PAC</strong>: Text PAC to 65075
            <br />
            For <strong>STAC</strong>: Text STAC to 75075
          </Paragraph>
          <Paragraph>Or visit your current network&apos;s website to get the code you need.</Paragraph>
        </Section>
      </div>
    </BlockContainer>
  )
}

export default PacStacExplained
