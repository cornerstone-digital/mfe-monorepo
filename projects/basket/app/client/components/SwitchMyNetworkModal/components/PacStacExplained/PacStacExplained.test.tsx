import PacStacExplained from './PacStacExplained'

describe('<PacStacExplained />', () => {
  test('should render', () => {
    const wrapper = shallow(<PacStacExplained />)
    expect(wrapper.length).toEqual(1)
    expect(snapshot(wrapper)).toMatchSnapshot()
  })
})
