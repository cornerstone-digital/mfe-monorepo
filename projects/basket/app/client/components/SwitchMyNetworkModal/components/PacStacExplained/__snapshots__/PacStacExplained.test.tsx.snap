// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PacStacExplained /> should render 1`] = `
<BlockContainer
  appearance="light1"
  borderAppearance="none"
  collapse={
    Object {
      "collapsable": true,
      "collapsed": true,
      "icon": Object {
        "appearance": "dark",
        "justify": "left",
      },
    }
  }
  guttering="md"
  heading={
    Object {
      "fontWeight": "bold",
      "size": 6,
      "text": "What are PAC and STAC codes?",
    }
  }
  marginBottom={2}
  theme={null}
>
  <div
    className="content"
  >
    <Icon
      appearance="dark"
      name="info-circle"
      responsive={true}
      size={2}
      theme={null}
    />
    <Section
      appearance="light1"
      borderAppearance="none"
      borderShadow={false}
      contentWidth="full"
      gutters="none"
      paddingLeft={2}
      roundedBorders={false}
      spacingOffsetVertical="none"
      theme={null}
      type="div"
    >
      <Heading
        appearance="inherit"
        focusable={false}
        fontWeight="bold"
        level={3}
        size={6}
        theme={null}
      >
        Use a PAC code to:
      </Heading>
      <List
        hideDecorators={false}
        items={
          Array [
            Object {
              "content": "Cancel your contract with your current network",
              "id": "1",
            },
            Object {
              "content": "Move your current mobile number over to Vodafone",
              "id": "2",
            },
            Object {
              "content": "Example code: ABC123456",
              "id": "3",
            },
          ]
        }
        theme={null}
        type="unordered"
      />
      <Heading
        appearance="inherit"
        focusable={false}
        fontWeight="bold"
        level={3}
        size={6}
        theme={null}
      >
        Use a STAC code to:
      </Heading>
      <List
        hideDecorators={false}
        items={
          Array [
            Object {
              "content": "Cancel your contract with your current network",
              "id": "4",
            },
            Object {
              "content": "Disconnect your current mobile number. You'll get a new number with Vodafone",
              "id": "5",
            },
            Object {
              "content": "Example code: 123456ABC",
              "id": "6",
            },
          ]
        }
        theme={null}
        type="unordered"
      />
      <Heading
        appearance="inherit"
        focusable={false}
        fontWeight="bold"
        level={3}
        size={6}
        theme={null}
      >
        How do I get my PAC or STAC?
      </Heading>
      <Paragraph
        theme={null}
      >
        Just send a free text to 65075 from the mobile you want to switch
      </Paragraph>
      <Paragraph
        theme={null}
      >
        For 
        <strong>
          PAC
        </strong>
        : Text PAC to 65075
        <br />
        For 
        <strong>
          STAC
        </strong>
        : Text STAC to 75075
      </Paragraph>
      <Paragraph
        theme={null}
      >
        Or visit your current network's website to get the code you need.
      </Paragraph>
    </Section>
  </div>
</BlockContainer>
`;
